spiketoolkit/comparison/groundtruthcomparison.py

```diff
@@ -0,0 +1,291 @@
+from .basecomparison import BaseComparison
+
+import numpy as np
+import pandas as pd
+
+
+# Note for dev,  because of  BaseComparison internally:
+#     sorting1 = gt_sorting
+#     sorting2 = other_sorting
+
+class GroundTruthComparison(BaseComparison):
+    """
+    Class to compare a sorter to ground truth.
+    """
+    
+    
+    def __init__(self, gt_sorting, other_sorting, gt_name=None, other_name=None,
+                delta_frames=10, min_accuracy=0.5, exhaustive_gt=False,
+                n_jobs=1, verbose=False):
+        BaseComparison.__init__(self, gt_sorting, other_sorting, sorting1_name=gt_name, sorting2_name=other_name,
+                                    delta_frames=delta_frames, min_accuracy=min_accuracy, n_jobs=n_jobs, verbose=verbose)
+        self.exhaustive_gt = exhaustive_gt
+
+        self._do_count()
+
+    
+    def _do_count(self):
+        """
+        Do raw count into a dataframe.
+        """
+        unit1_ids = self._sorting1.get_unit_ids()
+        columns = ['tp', 'fn', 'cl','fp', 'nb_gt', 'nb_other', 'other_id']
+        self.count = pd.DataFrame(index=unit1_ids, columns=columns)
+        for u1 in unit1_ids:
+            u2 = self._unit_map12[u1]
+            
+            self.count.loc[u1, 'tp'] = np.sum(self._labels_st1[u1] == 'TP')
+            self.count.loc[u1, 'cl'] = sum(e.startswith('CL') for e in self._labels_st1[u1])
+            self.count.loc[u1, 'fn'] = np.sum(self._labels_st1[u1] == 'FN')
+            self.count.loc[u1, 'nb_gt'] = self._labels_st1[u1].size
+            self.count.loc[u1, 'other_id'] = u2
+
+            if u2==-1:
+                self.count.loc[u1, 'fp'] = 0
+                self.count.loc[u1, 'nb_other'] = 0
+            else:
+                self.count.loc[u1, 'fp'] = np.sum(self._labels_st2[u2] == 'FP')
+                self.count.loc[u1, 'nb_other'] = self._labels_st2[u2].size
+            
```

Comment:
in the base sorting comparison I would add a `get_confusion_matrix()` function. This is useful for both the GroundTruthComparison and the SortingComparison and it will be usefool for the confusion matrix widget when we port it

Reply:
this comment was meant for the BaseSortingComparison

Reply:
OK